pkg/aws/actuator/actuator.go

```diff
@@ -326,47 +330,89 @@ func (a *AWSActuator) sync(ctx context.Context, cr *minterv1.CredentialsRequest)
 		logger.Debug("credentials already up to date")
 		return nil
 	}
-
-	credentialsRootSecret, err := a.GetCredentialsRootSecret(ctx, cr)
-	if err != nil {
-		logger.WithError(err).Error("issue with cloud credentials secret")
-		return err
-	}
-
-	switch credentialsRootSecret.Annotations[constants.AnnotationKey] {
-	case constants.InsufficientAnnotation:
-		msg := "cloud credentials insufficient to satisfy credentials request"
-		logger.Error(msg)
-		return &actuatoriface.ActuatorError{
-			ErrReason: minterv1.InsufficientCloudCredentials,
-			Message:   msg,
+	stsDetected, _ := utils.IsTimedTokenCluster(a.Client, logger)
+	if stsDetected {
+		if a.Codec == nil {
+			return fmt.Errorf("invalid codec, nil value")
 		}
-	case constants.PassthroughAnnotation:
-		logger.Debugf("provisioning with passthrough")
-		err := a.syncPassthrough(ctx, cr, credentialsRootSecret, logger)
+		logger.Info("actuator detected STS enabled cluster, enabling STS secret brokering for CredentialsRequests providing an IAM Role ARN")
+		awsSTSIAMRoleARN, err := awsSTSIAMRoleARN(a.Codec, cr)
 		if err != nil {
 			return err
 		}
-	case constants.MintAnnotation:
-		logger.Debugf("provisioning with cred minting")
-		err := a.syncMint(ctx, cr, logger)
+		if awsSTSIAMRoleARN != "" {
+			err = a.createSecret(awsSTSIAMRoleARN, cr.Spec.CloudTokenPath, cr.Spec.SecretRef.Name, cr.Spec.SecretRef.Namespace, logger)
```

Comment:
The check for this `err != nil` is outside the if statement.

Reply:
got fixed